File: Ios/denarii/denarii/user/DenariiUser.swift
//
//  DenariiUser.swift
//  denarii
//
//  Created by Andrew Katson on 9/30/23.
//

import Foundation

class DenariiUser {
    var resetID : Int = 0
    private(set) var reportID : String = ""
    private(set) var verificationReportStatus : String = ""
    private(set) var isVerified : Bool = false
    
    init(resetID: Int, reportID: String, verificationReportStatus: String, isVerified: Bool) {
        self.resetID = resetID
        self.reportID = reportID
        self.verificationReportStatus = verificationReportStatus
        self.isVerified = isVerified
    }
    
    init() {
        self.resetID = 0
        self.reportID = ""
        self.verificationReportStatus = ""
        self .isVerified = false
    }
}

File: Android/app/src/main/java/com/denarii/android/user/DenariiUser.java
package com.denarii.android.user;

import java.io.Serializable;

public class DenariiUser implements Serializable {
    private Integer resetID;
    private String reportID;
    private String verificationReportStatus;
    private boolean isVerified;

    public void setResetID(Integer newResetID) {
        resetID = newResetID;
    }

    public void setReportID(String newReportID) {
        reportID = newReportID;
    }

    public void setVerificationReportStatus(String newStatus) {
        verificationReportStatus = newStatus;
    }

    public void setVerified(boolean verified) {
        isVerified = verified;
    }

    public Integer getResetID() {
        return resetID;
    }

    public String getReportID() {
        return reportID;
    }

    public String getVerificationReportStatus() {
        return verificationReportStatus;
    }

    public boolean getIsVerified() {
        return isVerified;
    }
}
